### PR TITLE
Adding borrow modal to markets page

### PR DESCRIPTION
**Pull request by IanWoodard**
image

### CODE REVIEW

File: earn/src/components/lend/modal/BorrowModal.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  const [borrowAmountStr, setBorrowAmountStr] = useState<string>('');
+  const { activeChain } = useContext(ChainContext);
+
+  const selectedBorrow = selectedBorrows.find(
+    (borrow) => borrow.collateral.address === selectedCollateral.asset.address
+  );
+
+  const selectedLendingPair = selectedCollateral.matchingPairs.find(
+    (pair) =>
+      pair.token0.address === selectedBorrow?.asset.address || pair.token1.address === selectedBorrow?.asset.address
```
Comment (haydenshively) [inline earn/src/components/lend/modal/BorrowModal.tsx:63]: I've made this comment before, but don't we have a utility for this comparison?

File: earn/src/components/lend/modal/BorrowModal.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+  const maxBorrowAmount = useMemo(() => {
+    if (consultData === undefined || selectedBorrow === undefined) {
+      return null;
+    }
+    const sqrtPriceX96 = GN.fromBigNumber(consultData?.[1] ?? BigNumber.from('0'), 96, 2);
+    const nSigma = selectedLendingPair?.nSigma ?? 0;
+    const iv = consultData[2].div(1e6).toNumber() / 1e6;
+    let ltv = 1 / ((1 + 1 / ALOE_II_MAX_LEVERAGE + 1 / ALOE_II_LIQUIDATION_INCENTIVE) * Math.exp(nSigma * iv));
+    ltv = Math.max(0.1, Math.min(ltv, 0.9));
```
Comment (haydenshively) [inline earn/src/components/lend/modal/BorrowModal.tsx:86]: We do this a lot of different places now. Maybe time to make a util function?

File: earn/src/components/lend/modal/BorrowModal.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        .mul(sqrtPriceX96)
+        .setResolution(selectedBorrow.asset.decimals);
+    } else {
+      inTermsOfBorrow = inTermsOfBorrow
+        .div(sqrtPriceX96)
+        .div(sqrtPriceX96)
+        .setResolution(selectedBorrow.asset.decimals);
+    }
+    const maxBorrowSupply = GN.fromNumber(selectedBorrow.supply, selectedBorrow.asset.decimals);
+    const maxPotentialBorrow = inTermsOfBorrow.recklessMul(ltv);
```
Comment (haydenshively) [inline earn/src/components/lend/modal/BorrowModal.tsx:101]: These variable names don't make sense. Maybe `maxBorrowSupplyConstraint` and `maxBorrowHealthConstraint` or the same but "maxBorrowConstrainedBy____"

File: earn/src/components/lend/BorrowingWidget.tsx [hunk 4]
```diff
@@ -167,216 +168,232 @@ export default function BorrowingWidget(props: BorrowingWidgetProps) {
...
+                    const token0Color = tokenColors.get(account.token0.address);
+                    const token0Gradient = token0Color
+                      ? `linear-gradient(90deg, ${rgba(token0Color, 0.25)} 0%, ${GREY_700} 100%)`
+                      : undefined;
+                    const token1Color = tokenColors.get(account.token1.address);
+                    const token1Gradient = token1Color
+                      ? `linear-gradient(90deg, ${rgba(token1Color, 0.25)} 0%, ${GREY_700} 100%)`
+                      : undefined;
+                    return (
+                      <Fragment key={account.uniswapPool}>
```
Comment (haydenshively) [inline earn/src/components/lend/BorrowingWidget.tsx:209]: I don't think this key is sufficient. Would prefer it be a BorrowerNFT ID

File: earn/src/components/lend/BorrowingWidget.tsx [hunk 5]
```diff
@@ -167,216 +168,232 @@ export default function BorrowingWidget(props: BorrowingWidgetProps) {
...
+                    const token0Color = tokenColors.get(account.token0.address);
+                    const token0Gradient = token0Color
+                      ? `linear-gradient(90deg, ${GREY_700} 0%, ${rgba(token0Color, 0.25)} 100%)`
+                      : undefined;
+                    const token1Color = tokenColors.get(account.token1.address);
+                    const token1Gradient = token1Color
+                      ? `linear-gradient(90deg, ${GREY_700}  0%, ${rgba(token1Color, 0.25)} 100%)`
+                      : undefined;
+                    return (
+                      <Fragment key={account.uniswapPool}>
```
Comment (haydenshively) [inline earn/src/components/lend/BorrowingWidget.tsx:307]: Same

**Review by haydenshively — APPROVED** [review]

File: earn/src/components/lend/modal/BorrowModal.tsx (hunk 2 above)
Comment (IanWoodard) [reply earn/src/components/lend/modal/BorrowModal.tsx:86]: Coming up in the next PR

File: earn/src/components/lend/BorrowingWidget.tsx (hunk 4 above)
Comment (IanWoodard) [reply earn/src/components/lend/BorrowingWidget.tsx:210]: punting as we don't have that info yet

File: earn/src/components/lend/BorrowingWidget.tsx (hunk 5 above)
Comment (IanWoodard) [reply earn/src/components/lend/BorrowingWidget.tsx:309]: same as above

**Pull request merged**